contract/vm.go

```diff
@@ -891,17 +891,32 @@ func setRandomSeed(ctx *vmContext) {
 	ctx.seed = rand.New(randSrc)
 }
 
-func setContract(contractState *statedb.ContractState, contractAddress, payload []byte) ([]byte, []byte, error) {
+func setContract(contractState *statedb.ContractState, contractAddress, payload []byte, ctx *vmContext) ([]byte, []byte, error) {
 	codePayload := luacUtil.LuaCodePayload(payload)
 	if _, err := codePayload.IsValidFormat(); err != nil {
 		ctrLgr.Warn().Err(err).Str("contract", types.EncodeAddress(contractAddress)).Msg("deploy")
 		return nil, nil, err
 	}
 	code := codePayload.Code()
+
+	// if hardfork version 4
+	if ctx.blockInfo.ForkVersion >= 4 {
```

Comment:
I think this part would insure that code running in blocks with forkversion < 4 to run as previously @hayarobi

Reply:
The contract execution code was not modified. It remains exactly the same